search-ui/src/composables/entity-factory.ts

```diff
@@ -64,17 +64,32 @@ export const useEntity = () => {
     entity.status = newEntity.status
   }
 
-  const isBComp =  () => {
+  const isBComp = () => {
     return entity.legalType == 'BEN'
```

Comment:
Is this reactive? I'm thinking you might need to turn this into a computed value in case the entity is switched. Also can you use the enum for the check?